components/tor/renderer/onion_domain_throttle.cc

```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/tor/renderer/onion_domain_throttle.h"
+
+#include "net/base/net_errors.h"
+#include "net/base/url_util.h"
+#include "services/network/public/cpp/resource_request.h"
+
+namespace tor {
+
+OnionDomainThrottle::OnionDomainThrottle(
+    base::RepeatingCallback<bool()> read_is_tor_cb)
```

Comment:
pls replace this callback with an interface. We previously hit performance issues with `RepeatingCallback` in farbling code, because it's thread safe and is not fast when you `Run()` it a lot of times. Here it's called *for each navigation*, which can actually lead to performance regression in some scenarios.